homework/list.js

```diff
@@ -1,43 +1,53 @@
 'use strict';
 
-(function() {
+//the way of how to refactor list.js is the same as content.js 
+//repair some error in list.js
+// add promise into it
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+(function(exports) {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
+  var ListManager = function(){
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector("#note-content-wrapper");
+  }
+
+  ListManager.prototype = {
+    start() {
+      this.fetchList().then((function(data) {
```

Comment:
It's recommended to use Arrow function instead of calling `bind` API, since it makes code more concise. See:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions